Clarifications (#493)
rust-lang-nursery/reference: src/items/implementations.md

The nominal type is called the _implementing type_ and the associable items are
the _associated items_ to the implementing type.

Inherent implementations associate the contained items to the
implementing type.  Inherent implementations can contain [associated

> functions] (including methods), [associated types], and [associated

Reviewer: Associated types are type aliases, and as per the next statement, inherent impls cannot have them.

Author: Removed.

constants]. They cannot contain associated type aliases.

> [associated functions]: associated-items.html#associated-functions-and-methods

Reviewer: The link refs should be added to the bottom of the page in alphabetical (by ASCII) order.

[associated types]: associated-items.html#associated-types
[associated constants]: associated-items.html#associated-constants

> The path to an associated item is: any path to the implementing type,

Reviewer: It would be nice if the first "path" was a link to the chapter on paths. I wouldn't block the PR on it though.

Author: done I think.  Unless I linked to the wrong place.

followed by the path to the associate item within the inherent
implementation.

> Note that the path to the module containing the inherent

Reviewer: I'm not quite sure we want to have this note here. It's a restatement of the (visibility chapter)[https://doc.rust-lang.org/nightly/reference/visibility-and-privacy.html].

Author: That's fair.  I haven't read the whole reference yet, but it occurred to me as a subtle consequence that was implied by what was previously there.  Maybe a link to the visibility chapter would be more appropriate than restating?

Author: I removed the note paragraph and instead called it out in the example by showing side by side that impl blocks in different modules still use a path to the type.

implementation does not allow access to the associate item, unless the
implementing type is re-exported from the same location.

``` rust
pub mod color {
    pub struct Color(pub u8, pub u8, pub u8);
}

mod values {
    use super::color::Color;
    impl Color {
        pub fn red() -> Color {
            Color(255, 0, 0)
        }
    }
}

pub use self::color::Color;
fn main() {
    color::Color::red();  // actual path to the implementing type
    Color::red();  // rexported paths to the implementing type also work

>     // bright_theme::Color::red();  // Does not work, because use in `theme` is not pub

Reviewer: `bright_theme` doesn't exist. Did you mean `values` or did you want to rename module?

Author: whoops.  Changed the code sample too many times.  I was going to add another module with a pub use in it, but it ended up feeling like belaboring the point.

Author: fixed. and fixed the comment which mentioned yet another previous name for the secondary module.

The nominal type is called the _implementing type_ and the associable items are
the _associated items_ to the implementing type.

Inherent implementations associate the contained items to the
implementing type.  Inherent implementations can contain [associated
functions] (including methods) and [associated constants]. They cannot
contain associated type aliases.

> The [path] to an associated item has the following structure: any path

Reviewer 2: ```suggestion
The [path] to an associated item is any path
```

> to the implementing type, followed by the type path to the associate item

Reviewer 2: ```suggestion
to the implementing type, followed by the associated item's identifier as the final path component.
```

Type paths are a specific concept that I don't think is correct here.

implementation.

A type can also have multiple inherent implementations. An implementing type
must be defined within the same crate as the original type definition.

``` rust
pub mod color {
    pub struct Color(pub u8, pub u8, pub u8);

> 

Author: Reading the whole page again, I decided to remove the old example and just use the larger example I've written since (IMHO) it more clearly demonstrates what the prose above it is discussing.

    impl Color {
        pub const WHITE: Color = Color(255, 255, 255);
    }
}

mod values {
    use super::color::Color;
    impl Color {
        pub fn red() -> Color {
            Color(255, 0, 0)
        }
    }
}

pub use self::color::Color;
fn main() {

>     color::Color::WHITE; // actual path to the implementing type and impl in the same module

Reviewer 2: These comments end up requiring one to horizontally scroll to read them.  Can you move each one up to a line on its own?  Also, I would make them sentences (capitalize the first word).

    color::Color::red();  // impl blocks in different modules are still accessed through a path to the type

>     Color::red();  // rexported paths to the implementing type also work

Reviewer 2: "Re-exported"

the _associated items_ to the implementing type.

Inherent implementations associate the contained items to the implementing type.

> The path of an associated item consists of a path to the implementing type followed by

Reviewer 3: Hmm... I don't find the current nor the changed wording to be great. In particular "path" gets repeated twice in succession; can we find an alternate wording?

Author: I don't think we can really get away from repeating path here, since it is the proper descriptive word.  This sentence is defining how to construct the path to an item associated with a trait, and part of that path is the path to the implementing type.  IMO using a synonym for path just to avoid the repetition would be even less clear.

That being said, if you have a concrete suggestion on how to improve it I'm all for it :)

Reviewer 3: I got nothing... sorry :)
